bccsp/schemes/dlog/crypto/nymrh.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package idemix
+
+import (
+	math "github.com/IBM/mathlib"
+	"github.com/pkg/errors"
+)
+
+type NymRH []byte
+
+func (nym NymRH) AuditNymRh(
+	ipk *IssuerPublicKey,
+	rhAttr *math.Zr,
+	rhIndex int,
+	RNymRh *math.Zr,
+	curve *math.Curve,
+	t Translator,
+) error {
+	// Validate inputs
+	if ipk == nil {
+		return errors.Errorf("cannot verify idemix signature: received nil input")
```

Comment:
it's not an idemix signature we're verifying here, is it?

Reply:
This was based off of the same audit function in [`nymeid.go`](https://github.com/IBM/idemix/blob/9afbf3cf16847805693e05667d423e186844521f/bccsp/schemes/dlog/crypto/nymeid.go#L26), I assume that this function only gets called when attempting to verify / audit the idemix signature, I can change the error log if you like